### DISCUSSION TITLE
If fine tuning seemed to work without error and the loss went down does that mean I fine tuned?

### DISCUSSION THREAD

**Discussion by MotorCityCobra** [ggml-org/llama.cpp#5155] [Q&A]
The finetune section of the examples dir says the fine tuning only works on llama models. I ran finetune.exe for a few hours not on a llama model, but a quantized Mistral 7B model and the loss went down. No errors. By the end of it I had two guff files and a bin file. I've tried every combination of opening these files in koboldcpp and it usually will not load a model and crashes.  
If you could throw me a bone and tell me what the gguf files are for and what the one bin file is for when opening the model I'd appreciate it. Something is the lora adapters? The gguf files are just 125MB but the quantized model is about 6GB. So, does koboldcpp not allow me to run them because finetune.cpp really does not fine tune Mistral models and I haven't fine tuned anything, or do I just not know how to load the files?

If finetune.cpp cannot fine tune a Mistral 7B model where would I begin to modify the code so that it can?
Somewhere here on line 100? 

struct my_llama_lora_layer {
    // normalization
    struct ggml_tensor * attention_norm_a;
    struct ggml_tensor * attention_norm_b;

    // attention
    struct ggml_tensor * wq_a;
    struct ggml_tensor * wq_b;
    struct ggml_tensor * wk_a;
    struct ggml_tensor * wk_b;
    struct ggml_tensor * wv_a;
    struct ggml_tensor * wv_b;
    struct ggml_tensor * wo_a;
    struct ggml_tensor * wo_b;

    // normalization
    struct ggml_tensor * ffn_norm_a;
    struct ggml_tensor * ffn_norm_b;

    // ff
    struct ggml_tensor * w1_a;
    struct ggml_tensor * w1_b;
    struct ggml_tensor * w2_a;
    struct ggml_tensor * w2_b;
    struct ggml_tensor * w3_a;
    struct ggml_tensor * w3_b;
};

struct my_llama_lora {
    struct ggml_context * ctx = NULL;
    std::vector<uint8_t> data;

    my_llama_lora_hparams hparams;

    struct ggml_tensor * tok_embeddings_a;
    struct ggml_tensor * tok_embeddings_b;

    struct ggml_tensor * norm_a;
    struct ggml_tensor * norm_b;
    struct ggml_tensor * output_a;
    struct ggml_tensor * output_b;

    std::vector<my_llama_lora_layer> layers;
};

**Comment by BarfingLemurs** [comment]
use the export-lora binary to merge these:

`./export-lora -m model.gguf -l Lora -o merged-model.gguf`
